vector/src/main/java/im/vector/util/VectorRoomMediasSender.java

```diff
@@ -774,64 +774,31 @@ else if (null != imageSizes.mSmallImageSize) {
-                                                            aListener.onDone();
-                                                        }
-                                                    });
-                                                }
-                                            });
-
-                                            thread.setPriority(Thread.MIN_PRIORITY);
-                                            thread.start();
+                    if(mVectorRoomActivity.getResources().getBoolean(R.bool.show_image_compression_dialog)) {
+                        mImageSizesListDialog = new AlertDialog.Builder(mVectorRoomActivity)
+                                .setTitle(im.vector.R.string.compression_options)
+                                .setSingleChoiceItems(stringsArray, -1, new DialogInterface.OnClickListener() {
+                                    @Override
+                                    public void onClick(DialogInterface dialog, int which) {
+                                        final int fPos = which;
+                                        mImageSizesListDialog.dismiss();
+                                        sendFile(imageSizes, fPos, anImageUrl, filename, rotationAngle, roomMediaMessage, aListener);
+                                    }
+                                })
+                                .setOnCancelListener(new DialogInterface.OnCancelListener() {
+                                    @Override
+                                    public void onCancel(DialogInterface dialog) {
+                                        mImageSizesListDialog = null;
+                                        if (null != aListener) {
+                                            aListener.onCancel();
                                         }
-                                    });
-                                }
-                            })
-                            .setOnCancelListener(new DialogInterface.OnCancelListener() {
-                                @Override
-                                public void onCancel(DialogInterface dialog) {
-                                    mImageSizesListDialog = null;
-                                    if (null != aListener) {
-                                        aListener.onCancel();
                                     }
-                                }
-                            })
-                            .show();
+                                })
+                                .show();
+                    } else {
+                        // First position is the original image
+                        sendFile(imageSizes, 0, anImageUrl, filename, rotationAngle, roomMediaMessage, aListener);
+                    }
                 }
             } catch (Exception e) {
                 Log.e(LOG_TAG, "sendImageMessage failed " + e.getMessage(), e);
```

Comment:
Is this swallowing exceptions?  What is the user experience if there is an exception?  Silently failing isn't ideal.

Reply:
I have added a toast message.